### PR TITLE
pgsql: allow out of order parameters in startup message - v1

### PR DESCRIPTION
**Pull request by jufajardini**
This is a continuation of the work done by @catenacyber in https://github.com/OISF/suricata/pull/7778

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5579

Describe changes:
- support out of order parameters in the pgsql startup message
- move the database message into the optional parameters, as it's indeed an optional parameter
- update the json schema accordingly

There is still work to be done related to the initial bug report by Philippe (pgsql should not error out when reaching a parsing error, if it's still capable of finding the next PDU on the stream [5524](https://redmine.openinfosecfoundation.org/issues/5524))

suricata-verify-pr: 976

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 10158

File: etc/schema.json [hunk 1]
```diff
@@ -2717,11 +2714,23 @@
...
+                                            },
+                                            "options": {
+                                                "type": "string"
+                                            },
                                             "replication": {
                                                 "type": "string"
                                             }
                                         },
-                                        "additionalProperties": false
+                                        "additionalProperties": true
```
Comment (victorjulien) [inline etc/schema.json:2733]: is there a way to enforce that the type of the members is always `string`?

File: rust/src/pgsql/parser.rs [hunk 2]
```diff
@@ -2082,7 +2073,62 @@ mod tests {
             }
         }
 
-        // TODO keep adding more messages
+        //A series of response messages from the backend:
+        /* R   ·    S   ·application_name  S   ·client_encoding UTF8 S   ·DateStyle ISO, MDY S   &default_transaction_read_only off S   ·in_hot_standby off S   ·integer_datetimes on S   ·IntervalStyle postgres S   ·is_superuser off S   ·server_encoding UTF8 S   ·server_version 14.5 S   "session_authorization ctfpost S   #standard_conforming_strings on S   ·TimeZone Europe/Paris K      ···O··Z   ·I */
```
Comment (victorjulien) [inline rust/src/pgsql/parser.rs:2077]: formatting?

File: rust/src/pgsql/parser.rs [hunk 3]
```diff
@@ -2236,4 +2282,5 @@ mod tests {
 
         assert_eq!(ok_res, result);
     }
+
```
Comment (victorjulien) [inline rust/src/pgsql/parser.rs:2285]: nit: unrelated & unneeded change

File: etc/schema.json (hunk 1 above)
Comment (jufajardini) [reply etc/schema.json:2733]: For parameters status in postgresql, the values are always sent as string. Should we use something else?

File: rust/src/pgsql/parser.rs (hunk 2 above)
Comment (jufajardini) [reply rust/src/pgsql/parser.rs:2077]: /hides
VS Code made it look like it had line breaks. Am sorry for that >__<

File: rust/src/pgsql/parser.rs (hunk 3 above)
Comment (jufajardini) [reply rust/src/pgsql/parser.rs:2285]: roger!

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/8075

File: etc/schema.json (hunk 1 above)
Comment (jufajardini) [reply etc/schema.json:2733]: (sorry, I needed more coffee)

**Pull request closed without merging**